scheduler/src/cook/rest/api.clj

```diff
@@ -593,11 +593,7 @@
         image-config (:image docker)
         image (if (string? image-config)
                 image-config
-                (let [incremental-image-config (config-incremental/resolve-incremental-config uuid image-config)
-                      [resolved-config reason] (if incremental-image-config
-                                                 [incremental-image-config :resolved-incremental-config]
-                                                 ; use a fallback image in case there is a problem resolving an incremental config
-                                                 [(:image-fallback docker) :used-image-fallback])]
+                (let [[resolved-config reason] (config-incremental/resolve-incremental-config uuid image-config (:image-fallback docker))]
```

Comment:
Q: Should we resolve this at submit time or launch time? To me, it seems like we should resolve this at launch time --- otherwise, there's a lag. if we have a bad config, it bake it into a bunch of jobs that can't be fixed even if we fix the config. Versus making a config change and then all jobs submitted after the config change are good.  Thoughts?

Reply:
Resolving at launch time seems better, but the default image is rarely changed and errors can be mitigated with incremental rollout anyway. So it's not clear if it's worth the effort to refactor. (In either case it would not be part of this PR)

Reply:
Fair.